x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -297,7 +303,9 @@ public void checkIndexVersionThenExecute(final Consumer<Exception> consumer, fin
     public void prepareIndexIfNeededThenExecute(final Consumer<Exception> consumer, final Runnable andThen) {
         final State indexState = this.indexState; // use a local copy so all checks execute against the same state!
         // TODO we should improve this so we don't fire off a bunch of requests to do the same thing (create or update mappings)
-        if (indexState.indexExists && indexState.isIndexUpToDate == false) {
+        if (indexState == State.UNRECOVERED_STATE) {
+            consumer.accept(new ElasticsearchStatusException("State not yet recovered", RestStatus.SERVICE_UNAVAILABLE));
```

Comment:
I think a more vebose message would help users understand what's going on when this happens.
```suggestion
            consumer.accept(new ElasticsearchStatusException("Cluster state has not been recovered yet, cannot write to the security index", RestStatus.SERVICE_UNAVAILABLE));
```